src/client/app/components/ThreeDIntervalControlsComponent.tsx

```diff
@@ -0,0 +1,177 @@
+	const [numDaysCustom, setNumDaysCustom] = React.useState<number>(effectiveNumDays);
+	// True if custom numDays input is active.
+	const [showCustomNumDays, setShowCustomNumDays] = React.useState<boolean>(false);
+	// Define a flag to track if custom input is actively being used
+	const [isCustomInput, setIsCustomInput] = React.useState<boolean>(false);
+
+	// Keeps react-level state, and redux state in sync.
+	// Two different layers in state may differ especially when externally updated (chart link, history buttons.)
+	React.useEffect(() => {
+		// If user is in custom input mode, don't reset to standard options
+		if (!isCustomInput) {
+			const standardValues = Object.values(ThreeDNumDays)
+				.filter(v => typeof v === 'number' && v > 0) as number[];
+			const isCustom = !standardValues.includes(effectiveNumDays);
+			setShowCustomNumDays(isCustom);
+			setNumDaysCustom(effectiveNumDays);
+			setNumDays(isCustom ? CUSTOM_INPUT : effectiveNumDays.toString());
+		}
+	}, [numDaysRedux, effectiveNumDays, isCustomInput]);
+
+	// Returns true if this is a valid numDays.
+	const numDaysValid = (days: number): boolean => {
+		return Number.isInteger(days) && days >= MIN_DAYS && days <= MAX_DAYS;
+	};
+
+	// Updates values when the standard numDays menu is used.
+	const handleNumDaysChange = (value: string) => {
+		setIsCustomInput(false);
+		if (value === CUSTOM_INPUT) {
+			// Set menu value from standard value to special value to show custom
+			// and show the custom input area.
+			setShowCustomNumDays(true);
+			setNumDays(CUSTOM_INPUT);
+		} else {
+			// Set the standard menu value, hide the custom numDays input
+			// and numDays for graphing.
+			// Since controlled values know it is a valid integer.
+			setShowCustomNumDays(false);
+			updateNumDaysChange(Number(value));
+		}
+	};
+
+	// Updates value when the custom numDays input is used.
+	const handleCustomNumDaysChange = (value: number) => {
+		setIsCustomInput(true);
+		setNumDaysCustom(value);
+	};
+
+	const handleEnter = (key: string) => {
+		// This detects the enter key and then uses the previously entered custom
+		// numDays to set the numDays for the graphic.
+		if (key === 'Enter') {
+			updateNumDaysChange(numDaysCustom);
+		}
+	};
+
+	const updateNumDaysChange = (value: number) => {
+		// Update if okay value. May not be okay if this came from user entry in custom form.
+		if (numDaysValid(value)) {
+			dispatch(graphSlice.actions.updateThreeDNumDays(value));
+		}
+	};
+
+	const numDaysTranslations: Record<keyof typeof ThreeDNumDays, string> = {
+		OneMonth: '1.month',
+		SixMonths: '6.months',
+		OneYear: '1.year',
+		TwoYears: '2.years',
+		Max: '3.years',
+		Custom: 'custom.value'
+	};
+
+	return (
+		<div>
+			<div style={divTopBottomPadding}>
+				<p style={labelStyle}>
+					{translate('threeD.numDays')}:
+					<TooltipMarkerComponent page='home' helpTextId='help.home.threeD.numDays.tip' />
```

Comment:
There should be a placeholder in src/client/app/components/TooltipHelpComponent.tsx for this popup help.